### PR TITLE
Update to egui 0.23

### LINKED ISSUE #118: Update `renet_visualizer` to `egui` `0.23`

### PR DESCRIPTION
**Pull request by Zajozor**
Closes #118 

Ran into troubles with egui 0.23 and current renet version, so tried to fix it this way.
I'm not aware of the details of renet internals, so maybe I forgot something - on the surface it seems to do the job though.

Took the time to update examples as well.

### CODE REVIEW

File: demo_bevy/Cargo.toml [hunk 1]
```diff
@@ -18,10 +18,9 @@ steam = ["bevy_renet/steam", "steamworks"]
 bevy_rapier3d = "0.22.0"
 bevy = {version = "0.11", default-features = false, features = ["bevy_core_pipeline", "bevy_render", "bevy_asset", "bevy_pbr", "x11", "tonemapping_luts", "ktx2", "zstd"]}
 bevy_renet = { path = "../bevy_renet", features = ["serde"] }
-serde = { version = "1.0", features = [ "derive" ] }
+serde = { version = "1.0", features = ["derive"] }
```
Comment (Zajozor) [inline demo_bevy/Cargo.toml:21]: auto-fixed some of the linting - hope that's fine

File: demo_bevy/Cargo.toml [hunk 2]
```diff
@@ -18,10 +18,9 @@ steam = ["bevy_renet/steam", "steamworks"]
...
 bincode = "1.3.1"
-# Use version directly when egui is updated to 0.22
```
Comment (Zajozor) [inline demo_bevy/Cargo.toml:23]: I did not completely grasp what this comment means - so I just updated to bevy_egui 0.22 which (counter intuitively) uses egui 0.23

File: demo_chat/Cargo.toml [hunk 3]
```diff
@@ -8,10 +8,10 @@ edition = "2021"
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
 [dependencies]
-renet = { path = "../renet", features = [ "transport", "serde" ] }
+renet = { path = "../renet", features = ["transport", "serde"] }
 renet_visualizer = { path = "../renet_visualizer" }
-eframe = "0.22"
-serde = { version = "1.0", features = [ "derive" ] } 
+eframe = "0.23"
```
Comment (Zajozor) [inline demo_chat/Cargo.toml:13]: needed to use egui 0.23

File: renet_visualizer/src/lib.rs [hunk 4]
```diff
@@ -366,9 +366,11 @@ fn show_graph(
 
         let body = Shape::Rect(RectShape {
             rect,
-            rounding: Rounding::none(),
+            rounding: Rounding::ZERO,
             fill: Rgba::TRANSPARENT.into(),
             stroke: style.rectangle_stroke,
+            uv: Rect::ZERO,
+            fill_texture_id: egui::TextureId::Managed(0),
```
Comment (Zajozor) [inline renet_visualizer/src/lib.rs:373]: As far as I could tell this was the only "code change" that egui 0.22 -> 0.23 required

**Comment by ActuallyHappening** [comment]
Can we get this merged please?

**Comment by dennorak** [comment]
Would love to see this merged

File: demo_bevy/Cargo.toml (hunk 2 above)
Comment (lucaspoffo) [reply demo_bevy/Cargo.toml:23]: I forgot to remove this comment :0

**Comment by lucaspoffo** [comment]
Thanks!

**Pull request merged**